zebra-chain/src/orchard/arbitrary.rs

```diff
@@ -42,7 +42,10 @@ impl Arbitrary for note::Nullifier {
         use halo2::arithmetic::FieldExt;
 
         (any::<u64>())
-            .prop_map(|number| Self::from(pallas::Scalar::from_u64(number).to_bytes()))
+            .prop_map(|number| {
+                Self::try_from(pallas::Scalar::from_u64(number).to_bytes())
+                    .expect("a valid generated nullifier")
+            })
```

Comment:
```rust
(array::uniform32(any::<u8>()))
    .prop_map(|bytes| Self::try_from(bytes).expect("a valid generated nullifier"))
```

Reply:
> ```rust
> (array::uniform32(any::<u8>()))
>     .prop_map(|bytes| Self::try_from(bytes).expect("a valid generated nullifier"))
> ```

Hang on, doesn't this re-introduce the bug we're trying to fix?
Are all random 32-byte arrays valid pallas base points?

Reply:
It will generate invalid points that we don't want if we change to this suggestion. 

`pallas::Scalar::from_u64` will make the code always get a valid element:

https://github.com/zcash/pasta_curves/blob/main/src/arithmetic/fields.rs#L76

So i think that needs to stay.

Reply:
Unfortunately `from_u64` only gives us very small values of nullifiers 
![image](https://user-images.githubusercontent.com/552961/121741102-07ec5500-cacc-11eb-8f0e-c4c534cb1d69.png)

Reply:
If we add [this](https://github.com/ZcashFoundation/zebra/pull/2267/files#r650230937), we can use that here instead, stop panic'ing, and get a better distribution of `orchard::Nullifier`s

Reply:
You can use `FieldExt::from_bytes_wide` to implement `Arbitrary`. (Trait methods are automatically public.)
https://github.com/zcash/pasta_curves/blob/93df9c0cb9241e788916bca8e9a509b0b03f6023/src/fields/fq.rs#L715